DOC-9648: Separate the definitions of boolean and boolean field queries

abhinavdangeti opened the pull request:
+ Also fixing the example descriptions and formatting for-
    - boolean query
    - boolean field query

abhinavdangeti commented:
@RayOffiah This one's ready for review.

Also, separately - would you give us the right set of permissions to add reviewers for this PRs?

dnault commented on modules/fts/pages/fts-supported-queries-boolean-query.adoc:
```diff
@@ -0,0 +1,27 @@
+
+* *must_not*: Result documents should satisfy any of these queries
+
+* *should*: Result documents must not satisfy any of these queries
```
Are the descriptions of "should" and "must not" swapped?

Separate note: a disjunction query's "min" property works for both "should" and "must not", which lets a user express things like "MUST NOT satisfy [min] or more of these disjuncts", or "SHOULD satisfy at least [min] of these disjuncts". I don't know if it's worth documenting that here, but I found it interesting.

abhinavdangeti replied:
Shoot thanks.

Ya should look into updating the min part separately.

dnault reviewed (changes requested):
Nice improvement!

RayOffiah commented:
@abhinavdangeti <-- Could you fix the merge conflicts on this PR? Thanks

I'll arrange for you to be added to the set-reviewer list.

abhinavdangeti committed:
DOC-9648: Separate the definitions of boolean and boolean field queries

+ Also fixing the example descriptions and formatting for-
    - boolean query
    - boolean field query

abhinavdangeti commented:
@RayOffiah Done.
